pkgs/tools/misc/android-tools/default.nix

```diff
@@ -9,11 +9,11 @@ in
 
 stdenv.mkDerivation rec {
   pname = "android-tools";
-  version = "31.0.3p1";
+  version = "33.0.3";
 
   src = fetchurl {
     url = "https://github.com/nmeum/android-tools/releases/download/${version}/android-tools-${version}.tar.xz";
-    sha256 = "1f2svy381r798hjinrc2xiwz13gkkqxfill343zvv8jqkn8rzxhf";
+    sha256 = "sha256-jOF02reB1d69Ke0PllciMfd3vuGbjvPBZ+M9PqdnC8U=";
   };
 
   patches = [
```

Comment:
These patches don't apply anymore. You might need to check if it can be removed or if it needs to be refreshed.

Reply:
Ah oops, yes, the patches.
I could not really find out why they were introduced.
The comment claims it "breaks the build". The PR #131691 "fmt 8.x releases doesn't supported by platform-tools". But I can not see that android-tools was broken before the patch was added? At least not in the normal environment?

@musfay @primeos maybe you can explain if/why using the system's fmt library is still needed?

Reply:
IIRC that patch was never required (I think the comment was only meant to explain why we're using `fmt_7` and not `fmt` (8.x)). Using libraries from Nixpkgs is usually preferred over using vendored libraries (as the ones from Nixpkgs are easier to override, review/verify, already compiled, etc.). However, using Nixpkgs/system libraries also has drawbacks (requires more effort to maintain and makes it more likely for the package to break if the Nixpkgs library is updated (etc.)). Looking at https://github.com/NixOS/nixpkgs/pull/131691#pullrequestreview-716033011 I wasn't too keen on this change anyway (likely mostly due to the higher maintenance effort) so I'd simply remove that part now that reverting that patch doesn't work anymore (you could check the source-code though to see if there's a flag for this now - if not and if they still vendor fmt 7 then we can also drop the dependency on `fmt_7`).

Reply:
I was in somehow urgent need today and just removed the first patch and built / used fastboot for flashing a mobile device successful. Would you be able to update the PR otherwise I could make a propose? :)